### ISSUE TITLE
[iOS] - Wrong Layouting in HorizontalStackLayout when WidthRequest in child element is not set

### ISSUE THREAD

**Issue by MAUIoxo** [dotnet/maui#17475]
### Description

I have an ItemTemplate (ContentView) with the following elements.

```xaml
<SwipeView>

    <Border Style="{StaticResource BorderStyle}">
        
        <StackLayout>
            <Grid ColumnDefinitions="*, 60, 60, 70" Margin="0, 10, 0, 10">
                
                <Label Grid.Row="0" Grid.Column="0" Text="Name of the Item" VerticalOptions="CenterAndExpand" FontSize="14" Margin="5, 0, 0, 0" />

                <!-- Min Amount -->
                <HorizontalStackLayout Grid.Row="0" Grid.Column="1" Margin="10, 0, 10, 0">
                    <Entry x:Name="MinEntry" Text="5" Style="{StaticResource EntryStyle}" />
                    <Label Text="w" VerticalOptions="Center" FontSize="12" />
                </HorizontalStackLayout>

                <!-- Max Amount -->
                <HorizontalStackLayout Grid.Row="0" Grid.Column="2" Margin="10, 0, 0, 0">
                    <Entry x:Name="MaxEntry" Text="6" Style="{StaticResource EntryStyle}" />
                    <Label Text="w" VerticalOptions="Center" FontSize="12" />
                </HorizontalStackLayout>


                <!-- Optimal Amount -->
                <HorizontalStackLayout Grid.Row="0" Grid.Column="3" VerticalOptions="Center" HorizontalOptions="End">
                    <Label Text="7" FontSize="13" VerticalTextAlignment="Center"/>
                    <Label Text="w" VerticalOptions="Center" Margin="3" FontSize="12" />
                </HorizontalStackLayout>
                
            </Grid>

            <Grid HeightRequest="20" ColumnDefinitions="Auto, *, Auto">
                <HorizontalStackLayout Grid.Column="0" HorizontalOptions="StartAndExpand">
                    <Image Source="dotnet_bot.png" HeightRequest="15" WidthRequest="15" Margin="0, 0, 2, 5"/>
                    <Label Text="First: " Style="{StaticResource ItemSelectionDetailsLabelStyle}"/>
                    <Label Text="234" Style="{StaticResource ItemSelectionDetailsNumberLabelStyle}"/>
                </HorizontalStackLayout>

                <HorizontalStackLayout Grid.Column="1" HorizontalOptions="CenterAndExpand" Margin="10, 0, 0, 0">
                    <!--<Image Source="dotnet_bot.png" HeightRequest="15" WidthRequest="15" />-->
                    <Image Source="dotnet_bot.png" />
                    <Label Text="Second Label: " Style="{StaticResource ItemSelectionDetailsLabelStyle}"/>
                    <Label Text="456" Style="{StaticResource ItemSelectionDetailsNumberLabelStyle}"/>
                </HorizontalStackLayout>

                <HorizontalStackLayout Grid.Column="2" HorizontalOptions="EndAndExpand" Margin="10, 0, 0, 0">
                    <!--<Image Source="dotnet_bot.png" HeightRequest="15" WidthRequest="15" Margin="5, 0, 2, 2" />-->
                    <Image Source="dotnet_bot.png" />
                    <Label Text="Third Lbl: " Style="{StaticResource ItemSelectionDetailsLabelStyle}"/>
                    <Label Text="678" Style="{StaticResource ItemSelectionDetailsNumberLabelStyle}"/>
                </HorizontalStackLayout>
            </Grid>                
            
        </StackLayout>
    </Border>


    <SwipeView.RightItems>
        <SwipeItems>
            <SwipeItem Text="Remove" BackgroundColor="{StaticResource DarkOrange1}" />
        </SwipeItems>
    </SwipeView.RightItems>
    
</SwipeView>
```

On **Android** it is layouted correctly (different icon image sizes can be neglected) and looks like follows:

![grafik](https://github.com/dotnet/maui/assets/140683727/a3244928-e616-431c-a98a-d86e9c479c1e)

<br/>

On **iOS** the layout of the elements in the **HorizontalStackLayout** is totally screwed up and also it looks like my label with Text "**Second Label:**" is missing:

![grafik](https://github.com/dotnet/maui/assets/140683727/3c82f5be-8695-4233-8345-3bdbebb2c9ef)

### Steps to Reproduce

1. Use the attached minimal reproducible example and look at **View1ItemTemplate.xaml**
2. Open in Android Emulator **Pixel 5 - API 33 (Android 13.0 - API 33)**
3. Layouting looks pretty much what it should look like at the Bottom with Labels **"First: 234", "Second Label: 456", "Third Lbl: 678"** (well, icon image sizes are not so nice in this example, but anyway)
4. Open in iOS Simulator **iPhone 14 iOS 16.2** and look at the labels and Icons there => They are totally screwed up and also "Second Label: 456" is totally missing here

![grafik](https://github.com/dotnet/maui/assets/140683727/f263ac82-84ee-4e97-9c2c-6117ce97079f)

5. Try with commented out code in HorizontalStackLayout: ```<Image Source="dotnet_bot.png" HeightRequest="15" WidthRequest="15" />``` which seems to work then

### Link to public reproduction project repository

[https://github.com/MAUIoxo/WrongLayoutingWithoutWidthRequestApp_17475](https://github.com/MAUIoxo/WrongLayoutingWithoutWidthRequestApp_17475)

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

7.0.92

### Affected platforms

iOS

### Did you find any workaround?

A workaround is setting the a **WidthRequest** as in the following example ```<Image Source="dotnet_bot.png" HeightRequest="15" WidthRequest="15" />```. In the example in the description above this part was commented out so that you can reproduce more easily.

**Comment by skha83** [comment]
Sounds like the same problem this PR is trying to fix #17120 
From the amount of already linked issues in that PR I really hope this fix makes it into the final .NET 8 release

**Comment by MAUIoxo** [comment]
Might be, I don't want to say yes or no and would recommend they check also with regards to the PR you mentioned

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 2.0(8.0.0-rc.1.9171). Repro on iOS 16.4, not repro on Android 13.0-API33 with below Project:
[WrongLayoutingApp.zip](https://github.com/dotnet/maui/files/12697309/WrongLayoutingApp.zip)

![image](https://github.com/dotnet/maui/assets/97647558/36cc8018-ae80-4047-8998-c02647110045)

**Comment by MAUIoxo** [comment]
I apologize for this question, but what is the course of action towards resolving the issue - will it be addressed for the November release of .NET 8, or is this most likely scheduled for a much later update?

**Comment by hartez** [comment]
This will most likely be addressed in a .NET 8 servicing release, not in the November release.

**Comment by MAUIoxo** [comment]
I tried to verify and reproduce this issue under the same circumstances, which wasn't really possible for me. As time went on and there are newer versions of XCode and also newer Versions of the emulators, I couldn't really take the one I mentioned when this bug was created.

Nevertheless, I tried to reproduced the issue with **Visual Studio 17.10.0 Preview 4.0** using **iPhone 15 Pro Max iOS 17.2** and `<MauiVersion>8.0.60-ci.net8.24307.1</MauiVersion>`.

![grafik](https://github.com/dotnet/maui/assets/140683727/e496334e-d9e1-40e9-8b3f-fb52654af581)

<br/>

The little images still do not have the same size, but as I can see this was also in the Screenshots of the original issue.
It got a bit worse as these little images are also not cut off at the bottom which wasn't the case in the original issue.

**But:** it behaves the same on Android and this is not what the issue was about.

From this standpoint, I would recommend to close the issue as solved
